packages/labs/src/tag-input/tagInput.tsx

```diff
@@ -167,8 +170,10 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
         const { selectionEnd, value } = event.currentTarget;
         if (event.which === Keys.ENTER && value.length > 0) {
             // enter key on non-empty string invokes onAdd
-            this.setState({ inputValue: "" });
-            Utils.safeInvoke(this.props.onAdd, value);
+            const shouldClearInput = Utils.safeInvoke(this.props.onAdd, value);
+            if (shouldClearInput) {
```

Comment:
Let's add a switch in the examples for "Unique values" 👍 

Can also come as a follow up PR, happy to take care of this